pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTIfStatement.java

```diff
@@ -5,29 +5,79 @@
 
 package net.sourceforge.pmd.lang.java.ast;
 
+import net.sourceforge.pmd.annotation.InternalApi;
+
+
+/**
+ * Represents an {@code if} statement, possibly with an {@code else} statement.
+ *
+ * <pre>
+ *
+ * IfStatement ::= "if" "(" {@linkplain ASTExpression Expression} ")" {@linkplain ASTStatement Statement}
+ *                 ( "else" {@linkplain ASTStatement Statement} )?
+ *
+ * </pre>
+ */
 public class ASTIfStatement extends AbstractJavaNode {
 
     private boolean hasElse;
 
+
+    @InternalApi
+    @Deprecated
     public ASTIfStatement(int id) {
         super(id);
     }
 
+
+    @InternalApi
+    @Deprecated
     public ASTIfStatement(JavaParser p, int id) {
         super(p, id);
     }
 
+
+    @InternalApi
+    @Deprecated
     public void setHasElse() {
         this.hasElse = true;
     }
 
+
+    /**
+     * Returns true if this statement has an {@code else} clause.
+     */
     public boolean hasElse() {
         return this.hasElse;
     }
 
+
+    /**
+     * Returns the node that represents the guard of this conditional.
+     * This may be any expression of type boolean.
+     */
+    public ASTExpression getGuardExpressionNode() {
+        return (ASTExpression) jjtGetChild(0);
+    }
+
+
+    /**
+     * Returns the statement that will be run if the guard evaluates
+     * to true.
+     */
+    public ASTStatement getTrueAlternative() {
```

Comment:
I'm unsure about the names getTrueAlternative / getFalseAlternative. In Rhino (Javascript), they have named it "getThenPart" and "getElsePart". In JLS, they talk about statements (first statement and second statement).
Not so helpfull, either.

How about these names?

* `getCondition()`
* `getThenBranch()`
* `getElseBranch()`

Reply:
I renamed the methods to get the branches. I'm not sure about `getCondition`. It's certainly a better name, but all other statement nodes have their condition named `getGuardExpressionNode`. Maybe we should rename them all to `getCondition` ?

Reply:
Maybe, not sure. If we rename it, then in a different PR and we should use the same name in all the nodes. For now, it is consistent as it is.